ui/cypress/e2e/dashboardsIndex.test.ts

```diff
@@ -74,21 +74,41 @@ describe('Dashboards', () => {
       cy.createDashboardTemplate(id)
     })
 
-    cy.getByTestID('empty-dashboards-list')
-      .getByTestID('add-resource-dropdown--button')
-      .click()
-
-    cy.getByTestID('add-resource-dropdown--template').click()
-
+    cy.getByTestID('empty-dashboards-list').within(() => {
+      cy.getByTestID('add-resource-dropdown--button').click()
+      cy.getByTestID('add-resource-dropdown--template').click()
+    })
     cy.getByTestID('template--Bashboard-Template').click()
-
     cy.getByTestID('template-panel').should('exist')
-
     cy.getByTestID('create-dashboard-button').click()
-
     cy.getByTestID('dashboard-card').should('have.length', 1)
   })
 
+  it('keeps user input in text area when attempting to import invalid JSON', () => {
+    cy.getByTestID('page-header').within(() => {
+      cy.contains('Create').click()
+    })
+
+    cy.getByTestID('add-resource-dropdown--import').click()
+    cy.contains('Paste').click()
+    cy.getByTestID('textarea')
+      .click()
+      .type('this is invalid JSON')
+    cy.get('button[title*="Import JSON"]').click()
+    cy.getByTestID('textarea--error').should('have.length', 1)
+    cy.getByTestID('textarea').should($s =>
```

Comment:
We might want to change this test id to be more descriptive, textarea seems way too vague. Same with page header possibly too. 

Like this is good getByTestID('add-resource-dropdown--button').

Reply:
Likely, this test will change again in the future when a more fully-featured text editor is put in place rather than just a plain <TextArea> per @alexpaxton. When the new editor is put in place, we will be sure to use a more descriptive test ID.

Reply:
Actually, you're right. "textarea" is too generic. I have made the change.